cmd/report.go

```diff
@@ -0,0 +1,139 @@
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/elastic-package/internal/cobraext"
+	"github.com/elastic/elastic-package/internal/reportgenerator"
+	_ "github.com/elastic/elastic-package/internal/reportgenerator/generators" // register all report generators
+	"github.com/elastic/elastic-package/internal/reportgenerator/outputs"
+)
+
+const reportLongDescription = `Use this command to generate various reports relative to the packages. Currently, the following types of reports are available:
+
+#### Benchmark report for Github
+
+These report will be generated by comparing local benchmark results against ones from another benchmark run.
+The report will show performance differences between both runs.
+
+It is formatted as a Markdown Github comment to use as part of the CI results.
+`
+
+func setupReportsCommand() *cobraext.Command {
+	var reportTypeCmdActions []cobraext.CommandAction
+
+	cmd := &cobra.Command{
+		Use:   "report",
+		Short: "Generate reports",
+		Long:  reportLongDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.Println("Generate reports")
+
+			if len(args) > 0 {
+				return fmt.Errorf("unsupported report type: %s", args[0])
+			}
+
+			return cobraext.ComposeCommandActions(cmd, args, reportTypeCmdActions...)
+		}}
+
+	cmd.PersistentFlags().BoolP(cobraext.FailOnMissingFlagName, "m", false, cobraext.FailOnMissingFlagDescription)
+	cmd.PersistentFlags().BoolP(cobraext.ReportFullFlagName, "", false, cobraext.ReportFullFlagDescription)
+	cmd.PersistentFlags().StringP(cobraext.ReportOutputFlagName, "", string(outputs.OutputFile), cobraext.ReportOutputFlagDescription)
+
+	// add benchmark report creation subcommand
+	cmd.AddCommand(getBenchReportCommand())
+
+	return cobraext.NewCommand(cmd, cobraext.ContextPackage)
+}
+
+func getBenchReportCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "benchmark",
+		Short: "Generate a benchmark report",
+		Long:  "Generate a benchmark report comparing local results against ones from another benchmark run.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.Printf("Generate benchmark reports\n")
+
+			failOnMissing, err := cmd.Flags().GetBool(cobraext.FailOnMissingFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.FailOnMissingFlagName)
+			}
+
+			reportOutput, err := cmd.Flags().GetString(cobraext.ReportOutputFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.ReportOutputFlagName)
+			}
+
+			isFull, err := cmd.Flags().GetBool(cobraext.ReportFullFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.ReportFullFlagName)
+			}
+
+			srcPath, err := cmd.Flags().GetString(cobraext.BenchReportSrcPathFlagName)
```

Comment:
`src` and `target` refer to the benchmarks to compare, right? But later in the report they are titled "Previous" and "New", or "Old" and "New" in the code.
wdyt about calling them `previous` (or `old`) and `new` instead? I think that this way it would be more clear what is being compared, and the flags would be consistent with the result.